mono/utils/mono-state.c

```diff
@@ -20,12 +20,188 @@ extern GCStats mono_gc_stats;
 
 // For AOT mode
 #include <mono/mini/mini-runtime.h>
+#include <mono/utils/mono-threads-debug.h>
+#include <mono/utils/mono-merp.h>
 
 #ifdef TARGET_OSX
 #include <mach/mach.h>
 #include <mach/task_info.h>
 #endif
 
+#include <sys/param.h>
+#include <sys/sysctl.h>
+#include <fcntl.h>
+
+typedef struct {
+	const char *directory;
+	MonoSummaryStage level;
+} MonoSummaryTimeline;
+
+static const char *configured_timeline_dir;
+static MonoSummaryTimeline log;
+
+static void
+file_for_summary_stage (const char *directory, MonoSummaryStage stage, gchar *buff, size_t sizeof_buff)
+{
+	g_snprintf (buff, sizeof_buff, "%s%scrash_stage_%d", directory, G_DIR_SEPARATOR_S, stage);
+}
+
+gboolean
+mono_summarize_set_timeline_dir (const char *directory)
+{
+	if (directory) {
+		configured_timeline_dir = strdup (directory);
+		return g_ensure_directory_exists (directory);
+	} else {
+		configured_timeline_dir = NULL;
+		return TRUE;
+	}
+}
+
+gboolean
```

Comment:
You don't use the return value, so why not `void`?

Reply:
I made it void